src/compressor.js

```diff
@@ -0,0 +1,76 @@
+define(function (require) {
+	'use strict';
+
+	var p5sound = require('master');
+	var Effect = require('effect');
+  var CustomError = require('errorHandler');
+
+	p5.Compressor = function() {
+		Effect.call(this);
+
+		this.compressor = this.ac.createDynamicsCompressor();
+    this.input.connect(this.compressor);
+    this.compressor.connect(this.wet);
+	};
+
+	p5.Compressor.prototype = Object.create(Effect.prototype);
+
+	p5.Compressor.prototype.process = function(src, attack, knee, 
+                                      ratio, threshold, release) {
+		src.connect(this.input);
+
+		this.set(attack, knee, ratio, threshold, release);
+	};
+
+
+  p5.Compressor.prototype.set = function (attack, knee, 
+                                ratio, threshold, release) {
+
+    // var self = this;
+    // var errorTrace = new Error().stack;
+
+    if (attack) {this.attack(attack);}
+    if (knee) {this.knee(knee);}
+    if (ratio) {this.ratio(ratio);}
+    if (threshold) {this.threshold(threshold);}
+    if (release) {this.release(release);}
+  };
+
+  p5.Compressor.prototype.attack = function (attack){
+    if (attack) {this.compressor.attack.value = attack;}
```

Comment:
i think we do! i'm adding a time option now-- 
but, this is one of the chunks of code that appears many times in the library, I wonder if it would be a worthwhile future project to make a base class for functions that are implemented specifically for many different object in p5-sound